src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/RenamerProjectTreeActionHandler.cs

```diff
@@ -96,21 +96,24 @@ public override async Task RenameAsync(IProjectTreeActionHandlerContext context,
             }
 
             var documentRenameResult = await CodeAnalysis.Rename.Renamer.RenameDocumentAsync(oldDocument, newFileWithExtension);
-            bool errorsDetected = !documentRenameResult.ApplicableActions.Select(action => action.GetErrors()).Any();
+
             // Check errors before applying changes
-            if (errorsDetected)
+            foreach (var action in documentRenameResult.ApplicableActions)
             {
-                string failureMessage = GetFailureMessageCannotApply(documentRenameResult);
-                _userNotificationServices.ShowWarning(failureMessage);
-                return;
+                foreach (var e in action.GetErrors())
+                {
+                    return;
```

Comment:
If Roslyn ends up adding any errors, we never rename the file, which seems wrong.

Reply:
moved the 'return' after the file rename